pkgs/top-level/all-packages.nix

```diff
@@ -11817,6 +11817,13 @@ in
 
   nv-codec-headers = callPackage ../development/libraries/nv-codec-headers { };
 
+  nvidia-docker = callPackage ../applications/virtualization/nvidia-docker {
+    inherit (linuxPackages) nvidia_x11;
+    glibc = (glibc.overrideAttrs (oldAttrs: {
+      patches = oldAttrs.patches ++ [ ../development/libraries/glibc/ldconfig-patchelf.patch ];
```

Comment:
What's this patch for? Some comments would be appreciated. In any case you should move the apply of this patch inside the main nvidia-docker nix file, no need to do it here. Also no need to use `inherit (linuxPackages) nvidia_x11`, you can just take `linuxPackages` as an argument instead.

Reply:
This ldconfig patch is just a bug, and should probably be always applied, but that might break other things and would be a huge rebuild. But the linked glibc issue on sourceware might someday be looked at, so that might solve itself in the future.